### PR TITLE
Fill out the systemd-nspawn support

### PR DESCRIPTION
**Pull request by terminalmage**
So far I have added a couple additional functions to manage and get the
container state, as well as a set of functions to run commands within
containers, and interfaces to `machinectl pull-*`.

**Still to add (in a future PR):**
- Function to run salt-bootstrap in a container
- Function to clone a container.
- Functions for `machinectl copy-to` and `machinectl copy-from`

### CODE REVIEW

File: salt/modules/nspawn.py [hunk 1]
```diff
@@ -1,19 +1,44 @@
...
 Manage nspawn containers
+
+.. versionadded:: Beryllium
+
+`systemd-nspawn(1)`__ is a tool used to manage lightweight namespace
+containers. This execution module provides several functions to help manage
+these containers.
+
+.. __: http://www.freedesktop.org/software/systemd/man/systemd-nspawn.html
+
```
Comment (s0undt3ch) [inline salt/modules/nspawn.py]: Lovely!!!! Unnamed links! Go @terminalmage!

**Comment by thatch45** [comment]
Once tests pass we can merge in this iteration

**Comment by terminalmage** [comment]
Yeah I should be able to get the salt-bootstrap function (already partially written) in a separate PR.

**Comment by terminalmage** [comment]
Just pushed some lint fixes.

**Comment by terminalmage** [comment]
Also just pushed fixes for missing CLI examples caught in the Cent7 tests.

**Pull request merged**